sympy/stats/joint_rv_types.py

```diff
@@ -259,21 +259,23 @@ class MultinomialDistribution(JointDistribution):
     is_Discrete = True
 
     def check(self, n, p):
-        _value_check(((n > 0) != False) and isinstance(n, Integer),
+        _value_check(((n > 0) != False),
                         "number of trials must be a positve integer")
         for p_k in p:
-            _value_check((p_k >= 0) != False,
+            _value_check((p_k >= 0) != False and (p_k <= 1) != False,
                         "probability must be at least a positive symbol.")
+        _value_check(Eq(sum(p), 1) != False,
+                        "probabilities must sum to 1")
 
     @property
     def set(self):
-        return Range(0, self.n)**len(self.p)
+        return Interval(0, self.n)**len(self.p)
```

Comment:
wouldn't this have to be the intersection with Integers? But note that this currently fails:
```python
>>> (Interval(1,10)*Interval(3,7)).intersection(S.Integers)
Intersection(Integers, Interval(1, 10) x Interval(3, 7))
>>> list(_)
hangs
```

Reply:
Actually, I have used `Interval` just to make the things work. I want to use `Range` but it is unable to handle symbolic parameters. This is just a trick.

Reply:
trick or not, `Interval(0,3)**2` is not the same thing as `Range(4)**2`. That is why I suggest intersecting it with Integers.

Reply:
But then it will hang. I want to use `Range` with symbolic parameters. Otherwise it won't be possible to compute marginal distributions when `Interval` is intersected with `Integers`. I have raised the issue and mentioned you. Please take a look and let me know if it's possible to make the change in `__new__` of `Range`.

Reply:
Or I think can a bit of pre-processing of limits in `sympy.stats.join_rv` in `marginal_distributions`.

Reply:
You can try the change and see what breaks.

Reply:
CASE - 1
Change
======
```
@property
    def set(self):
        return Interval(0, self.n).intersection(S.Integers)**len(self.p)
```
Test Results
=========
```
============================= test process starts ==============================
executable:         /usr/bin/python3  (3.6.7-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.15.3
random seed:        4338608
hash randomization: on (PYTHONHASHSEED=3957755703)

sympy/stats/tests/test_joint_rv.py[10] ....E....f                         [FAIL]

________________________________________________________________________________
_____________ sympy/stats/tests/test_joint_rv.py:test_Multinomial ______________
Traceback (most recent call last):
  File "/home/gagandeep/sympy/sympy/stats/tests/test_joint_rv.py", line 68, in test_Multinomial
    marg = Sum(Piecewise((p1**x1*p2**C[1]*p3**C[2]*factorial(n)/(factorial(x1)*
  File "/home/gagandeep/sympy/sympy/stats/joint_rv.py", line 187, in __getitem__
    (self.name, self.pspace.component_count - 1))
ValueError: Index keys for C can only up to 0.

= tests finished: 8 passed, 1 expected to fail, 1 exceptions, in 10.89 seconds =
DO *NOT* COMMIT!
```
CASE - 2
Change
======
```
@property
    def set(self):
        return (Interval(0, self.n)**len(self.p)).intersection(S.Integers)
```
Test Results
=========
```
============================= test process starts ==============================
executable:         /usr/bin/python3  (3.6.7-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.15.3
random seed:        4338608
hash randomization: on (PYTHONHASHSEED=3957755703)

sympy/stats/tests/test_joint_rv.py[10] ....E....f                         [FAIL]

________________________________________________________________________________
_____________ sympy/stats/tests/test_joint_rv.py:test_Multinomial ______________
Traceback (most recent call last):
  File "/home/gagandeep/sympy/sympy/stats/tests/test_joint_rv.py", line 68, in test_Multinomial
    marg = Sum(Piecewise((p1**x1*p2**C[1]*p3**C[2]*factorial(n)/(factorial(x1)*
  File "/home/gagandeep/sympy/sympy/stats/joint_rv.py", line 187, in __getitem__
    (self.name, self.pspace.component_count - 1))
ValueError: Index keys for C can only up to 0.

= tests finished: 8 passed, 1 expected to fail, 1 exceptions, in 10.89 seconds =
DO *NOT* COMMIT!
```

Reply:
I am unable to figure out a way to making the things work except for using `Interval`. `Range` should be modified.